memory/conversation_summary.py

```
@@ -0,0 +1,109 @@
+    """
+    try:
+        # Get the last 20 messages for summarization
+        recent_messages = conversation_history[-20:]
+        
+        # Create a prompt for the summary
+        messages_text = "\n".join([
+            f"{msg['role']}: {msg['content']}"
+            for msg in recent_messages
+        ])
+        
+        prompt = f"""Please provide a concise summary of the following conversation, focusing on:
```

Ruff (0.8.2)

Local variable `prompt` is assigned to but never used

Remove assignment to unused variable `prompt`

(F841)